components/eam/src/physics/crm/physpkg.F90

```diff
@@ -1805,1229 +1180,654 @@ subroutine tphysac (ztodt,   cam_in,  &
+              clear_rh=mmf_clear_rh) ! clear air relative humidity for water uptake
+      call physics_update(state, ptend, ztodt, tend)
+
+      ! ! deep convective aerosol transport
+      ! call convect_deep_tend_2( state, ptend, ztodt, pbuf, &
+      !        mu, eu, du, md, ed, dp, dsubcld, jt, maxg,    &
+      !        ideep, lengath, species_class )
+      ! call physics_update(state, ptend, ztodt, tend)
+
+      ! ! check tracer integrals
+      ! call check_tracers_chng(state, tracerint, "cmfmca", nstep, ztodt,  zero_tracers)
 
-    call t_startf('bc_cld_diag_history_write')
+    end if
+  end if ! l_tracer_aero
 
-    call cloud_diagnostics_calc(state, pbuf)
+  !-----------------------------------------------------------------------------
+  ! Moist physical parameteriztions complete: 
+  ! send dynamical variables, and derived variables to history file
+  !-----------------------------------------------------------------------------
+  call t_startf('bc_history_write')
+  call diag_phys_writeout(state, cam_out%psl)
+  call diag_conv(state, ztodt, pbuf)
+  call t_stopf('bc_history_write')
 
-    call t_stopf('bc_cld_diag_history_write')
+  !-----------------------------------------------------------------------------
+  ! Write cloud diagnostics on history file
+  !-----------------------------------------------------------------------------
+  call t_startf('bc_cld_diag_history_write')
+  call cloud_diagnostics_calc(state, pbuf)
+  call t_stopf('bc_cld_diag_history_write')
 
-if (l_rad) then
-    !===================================================
-    ! Radiation computations
-    !===================================================
+  !-----------------------------------------------------------------------------
+  ! Radiation computations
+  !-----------------------------------------------------------------------------
+  if (l_rad) then
     call t_startf('radiation')
-
-
-    call radiation_tend(state,ptend, pbuf, &
-         cam_out, cam_in, &
-         cam_in%landfrac,landm,cam_in%icefrac, cam_in%snowhland, &
-         fsns,    fsnt, flns,    flnt,  &
-         fsds, net_flx,is_cmip6_volc, ztodt, clear_rh=mmf_clear_rh)
+    call radiation_tend(state,ptend, pbuf, cam_out, cam_in, &
+                        cam_in%landfrac, landm, cam_in%icefrac, cam_in%snowhland, &
+                        fsns, fsnt, flns, flnt, fsds, &
+                        net_flx, is_cmip6_volc, ztodt, clear_rh=mmf_clear_rh)
 
     ! Set net flux used by spectral dycores
     do i=1,ncol
-       tend%flx_net(i) = net_flx(i)
+      tend%flx_net(i) = net_flx(i)
     end do
     
-    ! don't add radiative tendency to GCM temperature in case of superparameterization
-    ! as it was added above as part of crm tendency.
-    if (use_MMF) ptend%s = 0.
+    ! For MMF, don't add radiative tendency to GCM as it was added in CRM
+    ptend%s = 0.
 
     call physics_update(state, ptend, ztodt, tend)
```

Comment:
Actually, I'm a little confused why we even call physics_update here, since `ptend%s = 0` because we apply the heating within the CRM. We should maybe remove the calls to `physics_update` and `check_energy_chng` altogether here.

Reply:
Good point... I can't think of any argument against this. I'll remove them and leave a comment block explaining what's happening.

Reply:
done.